app/src/main/java/com/lightricks/feedexercise/data/FeedRepository.kt

```diff
@@ -1,9 +1,40 @@
 package com.lightricks.feedexercise.data
 
+import com.lightricks.feedexercise.network.FeedApiService
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+import javax.inject.Inject
+import javax.inject.Singleton
+
 /**
  * This is our data layer abstraction. Users of this class don't need to know
  * where the data actually comes from (network, database or somewhere else).
  */
-class FeedRepository {
-    //todo: implement
-}
\ No newline at end of file
+interface FeedRepository {
+    fun getFeed(): Flow<List<FeedItem>>
+    suspend fun refreshFeed(): RefreshFeedResult
+}
+
+sealed class RefreshFeedResult {
+    object Success : RefreshFeedResult()
+    data class Error(val error: Throwable) : RefreshFeedResult()
+}
```

Comment:
Can not this be replaced by the kotlin `Result` object.

Reply:
I dont think that you can actually

Reply:
We can do:
```kotlin
val result: Result<Unit> = Result.success(Unit)
result.onSuccess {}
    .onFailure { error -> }
```